build/Dockerfile.builder

```diff
@@ -1,4 +1,4 @@
-FROM registry.access.redhat.com/ubi9/go-toolset:1.20 as builder
+FROM registry.access.redhat.com/ubi9/go-toolset as builder
```

Comment:
Does this always promise go 1.20 ?

Reply:
for the image always go 1.20, I suppose not.
as golang 1.21 is able to compile 1.20 with go.mod I suppose it's not a big deal?

Reply:
IMHO, we must keep the `go` version the same as on `go.mod` Try this on your local development env with both versions of `go` - `1.20` and `1.21` 

```
❯ go version
go version go1.21.7 linux/amd64

❯ go version
go version go1.21.7 linux/amd64

❯ git status
       modified:   go.mod
       modified:   go.sum

```

```
❯ go version
go version go1.20 linux/amd64

❯ go mod tidy
❯ git status
# .. clean - no changes
```

In `go 1.21`  .. there is diff to `go.mod and go.sum` which now means someone has to investigate if this change 
breaks / introduces any difference in kepler which IMHO is a wasted effort.

Reply:
ok, for this PR, I am going to make the builder with 1.20 tag. And apply our comments at https://github.com/sustainable-computing-io/kepler/discussions/1362#discussioncomment-9154786 
as the builder seems not due with kepler's CVE for now.
@rootfs , @sthaha , @vimalk78, @vprashar2929 , any other comments before I move to update this PR?

Reply:
Sounds good to me